src/style/style_layer/layer_properties.js.ejs

```diff
@@ -47,4 +47,6 @@ const paint: Properties<PaintProps> = new Properties({
 <% } -%>
 });
 
-export default { paint<% if (layoutProperties.length) { %>, layout<% } %> };
+export default ({ paint<% if (layoutProperties.length) { %>, layout<% } %> }: $Exact<{
+  paint: Properties<PaintProps><% if (layoutProperties.length) { %>, layout: Properties<LayoutProps><% } %>
+}>);
```

Comment:
Without adding this explicit annotation, Flow was inferring weaker types for these objects from their use in the constructor to `StyleLayer`, as `{layout?: Properties<...>, paint: Properties<...>}`

Reply:
👍 Add this as a comment.